pkg/agent/proxy/proxier.go

```diff
@@ -71,16 +75,24 @@ type proxier struct {
 	isIPv6       bool
 }
 
+func endpointKey(endpoint k8sproxy.Endpoint, protocol binding.Protocol) string {
+	return fmt.Sprintf("%s/%s", endpoint.String(), protocol)
+}
+
 func (p *proxier) isInitialized() bool {
 	return p.endpointsChanges.Synced() && p.serviceChanges.Synced()
 }
 
+// removeStaleServices removes all expired Services. Once a Service was deleted, all
```

Comment:
s/was deleted/is deleted
s/take response/takes responsibility
s/we don't need to do removeEndpoint/we don't need to call removeEndpoint

Reply:
Updated.